base/linalg/dense.jl

```diff
@@ -18,6 +20,8 @@ function scale!{T<:BlasFloat}(X::Array{T}, s::T)
     X
 end
 
+scale!{T<:BlasFloat}(s::T, X::Array{T}) = scale!(X, s)
```

Comment:
No they don't. We don't agree here and for those of us who use this method, it doesn't cause trouble.